### PR TITLE
Feat/116 user dashboard password settings page

### PR DESCRIPTION
**Pull request by Lftobs**
# Description

**Closes [#116](https://github.com/hngprojects/hng_boilerplate_remix/issues/116)

# Changes proposed

 Added dashboard password setting page

## What were you told to do?

User Dashboard Password Settings Page

## What did you do?
-  Develop the Password Settings page based on the provided design and requirements.
-  Implement a responsive layout that works well across different devices and browsers.
-  Create password input fields with mask/unmask functionality.
-  Implement client-side validation for password requirements.
-  Develop server-side logic for password update.
-  Implement error handling and success messages.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [ ] My code follows the code style of this project.
- [ ] This PR does not contain plagiarized content.
- [ ] The title and description of the PR is clear and explains the approach.
- [ ] I am making a pull request against the **dev branch** (left side).
- [ ] My commit messages styles matches our requested structure.
- [ ] My code additions will fail neither code linting checks nor unit test.
- [ ] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot from 2024-07-20 17-32-47-1

Screenshot from 2024-07-20 17-37-29

### CODE REVIEW

File: .vscode/settings.json
Comment (SirhmVFX) [inline .vscode/settings.json:1]: not allowed to edit settings.json

File: app/routes/dashboard.password-settings/route.tsx
Comment (SirhmVFX) [inline app/routes/dashboard.password-settings/route.tsx:1]: No side bar component

**Review by SirhmVFX — CHANGES_REQUESTED** [review]

File: app/routes/dashboard.password-settings/route.tsx
Comment (Lftobs) [reply app/routes/dashboard.password-settings/route.tsx:1]: its a component in issue [#5](https://github.com/hngprojects/hng_boilerplate_remix/issues/5#issue-2415567568l) and [#146](https://github.com/hngprojects/hng_boilerplate_remix/issues/148#issue-2419537228)

File: .vscode/settings.json
Comment (Lftobs) [reply .vscode/settings.json:1]: fixed

**Review by SirhmVFX — APPROVED** [review]

**Pull request merged**